### PR TITLE
Fix CompactMerkleTree when cloning an empty source tree.

### PR DESCRIPTION
**Pull request by AlCutter**

### CODE REVIEW

File: cpp/merkletree/merkle_tree_test.cc [hunk 1]
```diff
@@ -577,6 +577,12 @@ TEST_F(MerkleTreeTest, AddLeafHash) {
   EXPECT_EQ(kHashValue, tree.LeafHash(index));
 }
 
+TEST_F(CompactMerkleTreeTest, TestCloneEmptyTreeProducesWorkingTree) {
+  MerkleTree tree(new Sha256Hasher);
+  CompactMerkleTree compact(tree, new Sha256Hasher);
+  LOG(INFO) << compact.CurrentRoot();
```
Comment (pphaneuf) [inline cpp/merkletree/merkle_tree_test.cc]: Hmm, this should be some sort of `EXPECT_EQ`, no?

Comment (AlCutter) [inline cpp/merkletree/merkle_tree_test.cc]: It could be, but what I'm checking here is that the whole damn thing doesn't just `abort()` anymore.

**Comment by pphaneuf** [comment]
Done reviewing, the change to the code itself looks good, but the test needs a tweak?

File: cpp/merkletree/merkle_tree_test.cc (hunk 1 above)
Comment (pphaneuf) [inline cpp/merkletree/merkle_tree_test.cc]: Okay, that's a start, yeah, but what about something like this:

```C++
EXPECT_EQ(tree.CurrentRoot(), compact.CurrentRoot());
```

**Pull request merged**